pkg/collector/corechecks/cluster/orchestrator/processors/processor.go

```diff
@@ -165,16 +165,19 @@ func (p *Processor) Process(ctx *ProcessorContext, list interface{}) (processRes
 		})
 	}
 
-	// Split messages in chunks for orchestrator metadata and manifest data.
-	chunkCount := orchestrator.GroupSize(len(resourceMetadataModels), ctx.Cfg.MaxPerMessage)
-	metadataChunks := chunkResources(resourceMetadataModels, chunkCount, ctx.Cfg.MaxPerMessage)
-	manifestChunks := chunkResources(resourceManifestModels, chunkCount, ctx.Cfg.MaxPerMessage)
-	metadataMessages := make([]model.MessageBody, 0, chunkCount)
-	manifestMessages := make([]model.MessageBody, 0, chunkCount)
+	// Split messages in chunks
+	metadataChunker := &collectorOrchestratorChunker{}
+	chunkOrchestratorPayloadsBySizeAndWeight(resourceMetadataModels, resourceManifestModels, ctx.Cfg.MaxPerMessage, ctx.Cfg.MaxWeightPerMessageBytes, metadataChunker)
+	manifestChunker := &collectorOrchestratorChunker{}
+	chunkOrchestratorPayloadsBySizeAndWeight(resourceManifestModels, resourceManifestModels, ctx.Cfg.MaxPerMessage, ctx.Cfg.MaxWeightPerMessageBytes, manifestChunker)
+
+	chunkCount := len(metadataChunker.collectorOrchestratorList)
+	metadataMessages := make([]model.MessageBody, 0, len(metadataChunker.collectorOrchestratorList))
+	manifestMessages := make([]model.MessageBody, 0, len(metadataChunker.collectorOrchestratorList))
```

Comment:
shouldn't it be `manifestChunker` instead?

Reply:
As metadata chunking and manifest chunking both use serialized manifest size as indicator, the result of chunking is should be the same so `len(metadataChunker.collectorOrchestratorList)` should equal to `len(manifestChunker.collectorOrchestratorList)`. I can use `manifestChunker` to make code clearer